### ISSUE TITLE
Better error reporting in azure devops for sentinel builds

### ISSUE THREAD

**Issue by ashwinvenkatesha** [Azure/Azure-Sentinel#11129]
**Is your feature request related to a problem? Please describe.**
After adding content for ASIM parsers, azure devops pipeline complains that
`Template Id: ASimAuditEventIllumioSaaSCore is not valid in Line: 356 col: 25
Errors: The incomplete fragment is unexpected., Code: 'KS198', Severity: 'Error', Location: '31121..31122'
Expected: True
Actual:   False
`

From this message it isnt clear, which file this is pointing to. 
Also, what does location 'x..y' mean? What does the code KS198 mean?

Will be helpful if filename is present.

**Comment by v-sudkharat** [comment]
Hi @ashwinvenkatesha, Could you please let us know is that related to any PR? can you share PR no here.

**Comment by ashwinvenkatesha** [comment]
Hi @v-sudkharat 
https://github.com/Azure/Azure-Sentinel/pull/10994/checks?check_run_id=29130521606
this is from an older commit

**Comment by v-sudkharat** [comment]
@ashwinvenkatesha, The pipeline validations checks are pass for the above shared PR, can you check once.

**Comment by ashwinvenkatesha** [comment]
I understand the pipeline runs have passed for the pr in the recent runs
But not for that **run_id** isnt it.

If you open that link I shared above, you will see
```
Kqlvalidations.Tests.KqlValidationTests.Validate_ParsersFunctions_HaveValidKql(fileName: "ASimAuditEventIllumioSaaSCore.yaml", encodedFilePath: "L2hvbWUvdnN0cy93b3JrLzEvcy9QYXJzZXJzL0FTaW1BdWRpdE"...)

Template Id: ASimAuditEventIllumioSaaSCore is not valid in Line: 356 col: 25
                    Errors: The incomplete fragment is unexpected., Code: 'KS198', Severity: 'Error', Location: '31121..31122'
Expected: True
Actual:   False
```
This is the issue I am referring to, this error reporting makes no sense, since line 356, col 25 never had the error. 
The error was due to an extra semicolon at the end of the file, after many hours of frustration I realized the semicolon was not needed and that is when the next run went clean.
https://github.com/Azure/Azure-Sentinel/pull/10994/files/8eed458b0ee8b6e43bc4eb31035b500cb4b31495#diff-ae739a11b071ad0e82212c2ccc5a1063bbaec58629fd63e28f8de52ad4b31d03, check out ASimAuditEventIllumioSaaSCore.yaml, it has an extra semicolon at the end.
